pkg/es/client_test.go

```diff
@@ -0,0 +1,96 @@
+package es
+
+import (
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/elastic/go-elasticsearch/v7"
+	"github.com/esctl/esctl/pkg/config"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEsClient_New_ReturnsClient(t *testing.T) {
+	cfg := config.Cluster{
+		Name: "Hello",
+		Hosts: []string{
+			"http://hello1:9700",
+		},
+	}
+	c, err := New(&cfg)
+	assert.Nil(t, err, "Error should be nil")
+	assert.NotNil(t, c, "Client should not be nil")
+}
+
+func TestEsClient_New_Returns_err_when_host_missing(t *testing.T) {
+	cfg := config.Cluster{
+		Name:  "Hello",
+		Hosts: []string{},
+	}
+	c, err := New(&cfg)
+	assert.NotNil(t, err, "Error should be nil")
+	assert.Equal(t, errors.New("no hosts provided in config"), err)
+	assert.Nil(t, c, "Client should not be nil")
+}
+
+func TestESSearchClient_GetHealth(t *testing.T) {
+	resp := `
+	{
+		"active_primary_shards": 1,
+		"active_shards": 2,
+		"active_shards_percent_as_number": 100.0,
+		"cluster_name": "es-docker-cluster",
+		"delayed_unassigned_shards": 0,
+		"initializing_shards": 0,
+		"number_of_data_nodes": 3,
+		"number_of_in_flight_fetch": 0,
+		"number_of_nodes": 3,
+		"number_of_pending_tasks": 0,
+		"relocating_shards": 0,
+		"status": "green",
+		"task_max_waiting_in_queue_millis": 0,
+		"timed_out": false,
+		"unassigned_shards": 0
+	}
+	`
+	client := ElasticSearchClient{}
```

Comment:
Instantiate using New() in tests for better coverage? Also serves as a documentation that `ElasticSearchClient` should be instantiated using New() only. We can make the struct private if there's no need for it to be exposed outside.

Reply:
agree will make the changes.

Reply:
pushed the changes